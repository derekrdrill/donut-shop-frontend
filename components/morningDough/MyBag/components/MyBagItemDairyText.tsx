import * as React from 'react';
import { Typography } from '@mui/material';

import { MyBagItem } from '../../../../context/types/MyBagItem';

interface MyBagItemDairyTextProps {
  myBagItem: MyBagItem;
}

const MyBagItemDairyText = ({ myBagItem }: MyBagItemDairyTextProps) =>
  myBagItem.category === 'drink' && !myBagItem.soda && !myBagItem.bottled && myBagItem.dairy ? (
    <Typography data-testid='dairy-text' component='p' variant='caption'>
      {myBagItem.dairy?.replace(/([A-Z]+)*([A-Z][a-z])/g, '$1 $2').toUpperCase()}
    </Typography>
  ) : (
    <React.Fragment />
  );

export default MyBagItemDairyText;
