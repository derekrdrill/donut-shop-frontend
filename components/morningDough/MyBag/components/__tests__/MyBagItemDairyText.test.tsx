import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import MyBagItemDairyText from '../MyBagItemDairyText';

describe('MyBagItemDairyText unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(
        <MyBagItemDairyText
          myBagItem={{
            orderID: '1',
            menuItemID: 'icedCoffee',
            name: 'Iced Coffee',
            category: 'drink',
            subCategory: 'coldDrinks',
            quantity: 3,
            dairy: null,
            flavor: null,
            sugar: false,
            size: 'md',
            ice: 'normal',
            creamCheese: null,
            butter: null,
            bottled: false,
            soda: false,
            menuItemImage: 'icedCoffee.png',
          }}
        />,
      )
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders MyBagItemDairyText correctly (dairy-text)', async () => {
    render(
      <MyBagItemDairyText
        myBagItem={{
          orderID: '1',
          menuItemID: 'icedCoffee',
          name: 'Iced Coffee',
          category: 'drink',
          subCategory: 'coldDrinks',
          quantity: 3,
          dairy: 'halfAndHalf',
          flavor: null,
          sugar: false,
          size: 'md',
          ice: 'normal',
          creamCheese: null,
          butter: null,
          bottled: false,
          soda: false,
          menuItemImage: 'icedCoffee.png',
        }}
      />,
    );

    const menuItemButterText = await screen.findAllByTestId('dairy-text');

    expect(menuItemButterText[0]).toHaveClass('MuiTypography-root');
    expect(menuItemButterText[0]).toHaveTextContent('HALF AND HALF');
  });
});
