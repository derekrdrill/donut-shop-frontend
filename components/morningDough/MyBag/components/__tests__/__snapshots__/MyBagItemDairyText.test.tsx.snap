// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyBagItemDairyText unit tests renders as expected 1`] = `null`;
